frontend/src/app/shared/components/file-links/file-link-list/file-link-list.component.ts

```diff
@@ -131,14 +133,23 @@ export class FileLinkListComponent extends UntilDestroyedMixin implements OnInit
 
     this.fileLinks$ = this.fileLinkResourceService.collection(this.collectionKey);
 
+    this.currentUserService.isLoggedIn$
+      .pipe(this.untilDestroyed())
+      .subscribe((isLoggedIn) => { this.isLoggedIn = isLoggedIn; });
+
     this.fileLinks$
       .pipe(this.untilDestroyed())
       .subscribe((fileLinks) => {
         if (isNewResource(this.resource)) {
           this.resource.fileLinks = { elements: fileLinks.map((a) => a._links?.self) };
         }
 
-        this.storageInformation.next(this.instantiateStorageInformation(fileLinks));
+        // ToDo: Replace with anonymous user capabilities check.
+        // https://community.openproject.org/projects/openproject/work_packages/44850/activity
+        if (this.isLoggedIn) {
```

Comment:
I'd rather put that logic into the `instantiateStorageInformation()` method, to not split the display logic of the box to two different places.

Reply:
I'd argue this should not be set as an instance variable at all, instead we should use `combineLatest` and stay within the observable context

Reply:
I think it would make the already complex observable chain even more complex to understand. But it is possible. But have a look at the current state. What do you think?